plugins/module_utils/user.py

```diff
@@ -348,6 +362,10 @@ def user_mod(cursor, user, host, host_all, password, encrypted,
                     # Mysql and MariaDB differ in naming pam plugin and syntax to set it
                     if plugin in ('pam', 'ed25519'):
                         query_with_args = "ALTER USER %s@%s IDENTIFIED WITH %s USING %s", (user, host, plugin, plugin_auth_string)
+                    elif salt:
+                        if plugin in ['caching_sha2_password', 'sha256_password']:
+                            generated_hash_string = mysql_sha256_password_hash_hex(password=plugin_auth_string, salt=salt)
+                        query_with_args = ("ALTER USER %s@%s IDENTIFIED WITH %s AS 0x" + generated_hash_string), (user, host, plugin)
```

Comment:
wrong indentation?
Adding a new auth plugin that supports salt, will make line 368 to fail because `generated_hash_string` will be undefined.

Reply:
yes it's in the case we have hash generation for others plugins, I will put back the else with the fail, but to me, `generated_hash_string` will stay after the if else

Reply:
@Aohzan The test is already done at the module level. No need to repeat that test again here.
Alternatively, you can add a test `if not generated_hash_string:` and add a fail_json('unsupported auth plugin') ?

Reply:
Oh I see now that you already did revert back the changes @Aohzan. Thanks. I'm going out for lunch and will probably merge later.